tests/scripts/generate_test_driver.py

```diff
@@ -0,0 +1,181 @@
+                        help="Generate a file to be included from a CMakeLists.txt.\n"
+                        "The file defines CMake list variables with the script's\n"
+                        "inputs/outputs files. If FILE is omitted, the output \n"
+                        "name defaults to '<DRIVER>-list-vars.cmake'.")
+
+    args = parser.parse_args()
+
+    if not build_tree.looks_like_tf_psa_crypto_root("."):
+        raise RuntimeError("This script must be run from TF-PSA-Crypto root.")
+
+    builtin = Path("drivers/builtin")
+    if not builtin.is_dir():
+        raise RuntimeError("Do not find expected drivers/builtin directory.")
+    src_relpaths = get_src_relpaths(builtin, EXCLUDE_FILES)
+
+    dst_dir = Path(args.dst_dir)
+    if not dst_dir.is_dir():
+        raise RuntimeError(f"Do not find expected {args.dst_dir} directory.")
+
+    if args.list_vars_for_cmake:
+        fname = f"{args.driver}-list-vars.cmake" \
+                if args.list_vars_for_cmake == "__AUTO__" else args.list_vars_for_cmake
+
+        with open(dst_dir / fname, "w") as f:
+            f.write(f"set({args.driver}_input_files " + \
+                     " ".join(str(path) for path in src_relpaths) + ")\n\n")
+            f.write(f"set({args.driver}_files " + \
+                    " ".join(str(path) \
+                    for path in get_dst_relpaths(src_relpaths, args.driver)) + ")\n\n")
+            f.write(f"set({args.driver}_src_files " + \
+                    " ".join(str(path) \
+                    for path in src_relpaths if path.suffix == ".c") + ")")
+        return
+
+    #Step 1: Build the test driver tree from `drivers/builtin`
+    generator = test_driver.TestDriverGenerator(dst_dir, args.driver)
+    generator.build_tree(builtin, EXCLUDE_FILES)
+
+    #Step 2: Prefix the identifiers exposed by the built-in driver.
+    identifiers = generator.get_identifiers_with_prefixes(IDENTIFIER_PREFIXES)
+    external_identifiers = get_external_identifiers()
+    identifiers.difference_update(external_identifiers)
+
+    # MBEDTLS_ECP_LIGHT is defined in tf-psa-crypto/private/crypto_adjust_config_support.h
+    # if MBEDTLS_PK_PARSE_EC_EXTENDED or MBEDTLS_PK_PARSE_EC_COMPRESSED is enabled.
+    # It thus appears in 'external_identifiers' but is a built-in driver identifier
+    # that should be renamed thus force its renaming.
+    identifiers.add("MBEDTLS_ECP_LIGHT")
+
+    # MBEDTLS_PSA_ACCEL_ are not defined in the code base. They are supposed to
+    # be passed as extra configuration options. We want to prefix them,
+    # especially in 'crypto_adjust_config_enable_builtins.h', thus deduce them
+    # from the PSA_WANT_ ones and add them to the list of identifiers to
+    # prefix in the test driver code.
+    for identifier in external_identifiers:
+        if identifier.startswith("PSA_WANT_"):
+            identifiers.add(identifier.replace("PSA_WANT_", "MBEDTLS_PSA_ACCEL_", 1))
+
+    generator.prefix_identifiers(identifiers)
```

Comment:
Having reviewed `generate_test_driver.py`, I now think that all the code should be in the shared module `test_driver.py`, except:

* Branch-specific data: `EXCLUDE_FILES`, maybe `IDENTIFIER_PREFIXES`.
* The ad hoc fiddling of `identifiers` after adding the external identifiers and before calling `prefix_identifers`.
* Calling `main`.

I think this would be best handled by putting all the logic in `TestDriverGenerator`, and having a method to adjust the identifiers which does nothing by default and is overridden in a derived class defined in the branch-specific script.